app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -615,22 +616,42 @@ private void refreshView() {
     @Override
     public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
         // Retrieve the app record corresponding to the app selected
-        ApplicationRecord r = appRecordDropdownList.get(position);
+        String appId = appIdDropdownList.get(position);
 
-        // Set the id of the last selected app
-        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
-        prefs.edit().putString(KEY_LAST_APP, r.getUniqueId()).commit();
+        boolean appChanged = !appId.equals(CommCareApplication._().getCurrentApp().getUniqueId());
+        if (appChanged) {
+            // Set the id of the last selected app
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
+            prefs.edit().putString(KEY_LAST_APP, appId).commit();
 
-        // Refresh UI for potential new language
-        loadFields(false);
+            // Launch the activity to seat the new app
+            Intent i = new Intent(this, SeatAppActivity.class);
+            i.putExtra(KEY_APP_TO_SEAT, appId);
+            this.startActivityForResult(i, SEAT_APP_ACTIVITY);
+        }
+    }
 
-        // Initialize the selected app
-        CommCareApplication._().initializeAppResources(new CommCareApp(r));
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        switch(requestCode) {
+            case SEAT_APP_ACTIVITY:
+                if (resultCode == RESULT_OK) {
+                    refreshForNewApp();
+                }

```

Comment:
Personal preference of mine: it's nice to replace switch statements with one or two clauses with an if/else if.